### PR TITLE
Functional test for Xamarin.Android TrackEvent

### PR DESCRIPTION
**Pull request by guperrot**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the Windows code?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Add partial trackEvent functional test that works on Android.

[AB#72920](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/72920)

### CODE REVIEW

File: Tests/Microsoft.AppCenter.Test.Functional/Analytics/TrackEventTest.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+            // Test TrackEvent.
+            Analytics.TrackEvent("Hello World", properties);
+
+            // Wait for processing event.
+            await httpNetworkAdapter.HttpResponseTask;
+
+            // Verify.
+            Assert.Equal("POST", httpNetworkAdapter.Method);
+            var actualEventName = (string)httpNetworkAdapter.JsonContent["logs"][0]["name"];
```
Comment (guperrot) [inline Tests/Microsoft.AppCenter.Test.Functional/Analytics/TrackEventTest.cs]: Usually we split this into multiple asserts as many things can fail with an exception here such as null reference, type cast or array out of bounds.

Comment (jaelim-ms) [reply Tests/Microsoft.AppCenter.Test.Functional/Analytics/TrackEventTest.cs]: We filter json in HttpNetworkAdapter so `JsonContent["logs"]` is always nonnull, `JsonContent["logs"][0]` is always not empty.

File: SDK/AppCenter/Microsoft.AppCenter/AppCenter.cs [hunk 2]
```diff
@@ -64,5 +64,9 @@ static void PlatformStart(string appSecret, params Type[] services)
         static void PlatformSetCustomProperties(CustomProperties customProperties)
         {
         }
+
+        internal static void PlatformUnsetInstance()
```
Comment (guperrot) [inline SDK/AppCenter/Microsoft.AppCenter/AppCenter.cs]: We need this method on iOS and Windows.Shared as well to fix the build

Comment (jaelim-ms) [reply SDK/AppCenter/Microsoft.AppCenter/AppCenter.cs]: Added.

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs [hunk 3]
```diff
@@ -15,431 +15,436 @@
...
-        }
-
-        #endregion
-    }
+	/// <summary>
+	/// SDK core used to initialize, start and control specific service.
+	/// </summary>
+	public partial class AppCenter
+	{
+		// Internals for testing
```
Comment (guperrot) [inline SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs]: Formatting is wrong, was that edited on Mac? Should redo the patch or clean it up.

Comment (guperrot) [reply SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/AppCenter.cs]: Redid the change on windows, fixed in 2be10e

**Pull request merged**